sql/core/src/main/scala/org/apache/spark/sql/execution/r/MapPartitionsRWrapper.scala

```diff
@@ -25,6 +25,21 @@ import org.apache.spark.sql.Row
 import org.apache.spark.sql.types.{BinaryType, StructField, StructType}
 
 /**
+ * A function wrapper that applies the given R function to each partition of each group.
+ */
+private[sql] case class MapGroupsRWrapper(
```

Comment:
gapply for mutiple columns with RelationalGroupedDataset -> flatMap ...  seems to work fine. 
Will address other comments and push the changes any time soon.